File: boj/11047/11047.py
# https://www.acmicpc.net/problem/11047
# https://codeyoma.github.io/Computer-Science/1-Foundations--and--Theory/Algorithms/ps/boj/11047/11047
# ------------------------------------------------------------------------------
import sys

sys.setrecursionlimit(10**9)
input = sys.stdin.readline
output = sys.stdout.write

# ------------------------------------------------------------------------------
#                      /$$             /$$     /$$
#                     | $$            | $$    |__/
#   /$$$$$$$  /$$$$$$ | $$ /$$   /$$ /$$$$$$   /$$  /$$$$$$  /$$$$$$$
#  /$$_____/ /$$__  $$| $$| $$  | $$|_  $$_/  | $$ /$$__  $$| $$__  $$
# |  $$$$$$ | $$  \ $$| $$| $$  | $$  | $$    | $$| $$  \ $$| $$  \ $$
#  \____  $$| $$  | $$| $$| $$  | $$  | $$ /$$| $$| $$  | $$| $$  | $$
#  /$$$$$$$/|  $$$$$$/| $$|  $$$$$$/  |  $$$$/| $$|  $$$$$$/| $$  | $$
# |_______/  \______/ |__/ \______/    \___/  |__/ \______/ |__/  |__/
# solution ---------------------------------------------------------------------


def solve():
    return

# lgtm -------------------------------------------------------------------------
#  /$$        /$$$$$$  /$$$$$$$$ /$$      /$$
# | $$       /$$__  $$|__  $$__/| $$$    /$$$
# | $$      | $$  \__/   | $$   | $$$$  /$$$$
# | $$      | $$ /$$$$   | $$   | $$ $$/$$ $$
# | $$      | $$|_  $$   | $$   | $$  $$$| $$
# | $$      | $$  \ $$   | $$   | $$\  $ | $$
# | $$$$$$$$|  $$$$$$/   | $$   | $$ \/  | $$
# |________/ \______/    |__/   |__/     |__/
# ------------------------------------------------------------------------------


def log(*args):
    print(*args, file=sys.stderr)


def run_test(problem_number, test_number):

    input_path = f"{problem_number}/test-input-{test_number}.txt"
    output_path = f"{problem_number}/my-output-{test_number}.txt"

    import os
    if not os.path.exists(input_path):
        log("Input file not found:", input_path)
        return

    sys.stdin = open(input_path, 'r')
    sys.stdout = open(output_path, 'w')
    log(f"test case -  {test_number}")
    solve()
    print()


if __name__ == "__main__":
    args = sys.argv
    if len(args) == 3:
        p, t = map(int, args[1:])
        for i in range(1, t + 1):
            run_test(p, i)
    elif len(args) == 4:
        p, t, target = map(int, args[1:])
        run_test(p, target)
    else:
        solve()

File: boj/11047/11047.cpp
// https://www.acmicpc.net/problem/11047
// https://codeyoma.github.io/Computer-Science/1-Foundations--and--Theory/Algorithms/ps/boj/11047/11047
#include <iostream>
using namespace std;

#ifdef LOCAL
#    define LOG clog
#else
struct nullstream : ostream {
    nullstream()
        : ostream(nullptr) {}
};
nullstream LOG;
#endif

//--------------------------------------------------------------------------------------------------

#define MAX (1234567891)
#define MIN (-1234567891)

#include <iostream>
#include <vector>

int main() {
    int n, k;

    cin >> n >> k;
    vector<int> w(n);

    for (int i = 0; i < n; ++i) {
        cin >> w[i];
    }

    int cnt           = 0;
    int last_coin_pos = n - 1;

    while (k > 0 && last_coin_pos >= 0) {
        if ((k / w[last_coin_pos]) >= 1) {
            cnt += (k / w[last_coin_pos]);
            k %= w[last_coin_pos];
        }
        last_coin_pos--;
    }
    cout << cnt;
}

void _old_solution() {
    int n, k;
    cin >> n >> k;
    vector<int> coins(n);
    for (int i = 0; i < n; ++i) {
        cin >> coins[i];
    }

    int count = 0;
    int pos   = n - 1;
    while (k || pos >= 0) {
        if ((k / coins[pos]) >= 1) {
            count += (k / coins[pos]);
            k %= coins[pos];
        }
        pos--;
    }

    cout << count;
}